modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ExchangeContext.java

```diff
@@ -63,10 +67,16 @@
     public ExchangeContext(boolean crd, GridDhtPartitionsExchangeFuture fut) {
         int protocolVer = exchangeProtocolVersion(fut.firstEventCache().minimumNodeVersion());
 
+        boolean allNodesSupportsPmeFreeSwitch = allNodesSupports(fut.firstEventCache().allNodes(), PME_FREE_SWITCH);
+
+        if (!allNodesSupportsPmeFreeSwitch)
+            log.warning("PME-Free did not happen because the cluster is not ready. " +
+                "Not all nodes supports IgniteFeatures." + PME_FREE_SWITCH +".");
```

Comment:
Incorrect statement on node join, activate, etc.
"Current topology does not support pme-free switch because ..." is a correct statement.